src/main/java/com/conveyal/datatools/manager/models/transform/ReplaceFileFromStringTransformation.java

```diff
@@ -1,9 +1,56 @@
 package com.conveyal.datatools.manager.models.transform;
 
 import com.conveyal.datatools.common.status.MonitorableJob;
+import com.conveyal.datatools.manager.models.TableTransformResult;
+import com.conveyal.datatools.manager.models.TransformType;
+
+import java.io.ByteArrayInputStream;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.FileSystem;
+import java.nio.file.FileSystems;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardCopyOption;
 
 /**
  * This feed transformation will replace a file in the target zip (table) with the provided csv data.
  */
-public class ReplaceFileFromStringTransformation extends StringTransformation {
+public class ReplaceFileFromStringTransformation extends ZipTransformation {
+
+    public static ReplaceFileFromStringTransformation create(String csvData, String table) {
+        ReplaceFileFromStringTransformation transformation = new ReplaceFileFromStringTransformation();
+        transformation.csvData = csvData;
+        transformation.table = table;
+        return transformation;
+    }
+
+    @Override
+    public void validateParameters(MonitorableJob.Status status) {
+        if (csvData == null) {
+            status.fail("CSV data must not be null (delete table not yet supported)");
+        }
+    }
+
+    @Override
+    public void transform(FeedTransformZipTarget zipTarget, MonitorableJob.Status status) {
+        String tableName = table + ".txt";
+        // Run the replace transformation
+        Path targetZipPath = Paths.get(zipTarget.gtfsFile.getAbsolutePath());
+        try(FileSystem targetZipFs = FileSystems.newFileSystem(targetZipPath, (ClassLoader) null)) {
```

Comment:
Space after `try`.

Reply:
Fixed in https://github.com/ibi-group/datatools-server/pull/589/commits/ee203531500e572fcb51496610a90f34ddd5e48e